src/qt/optionsdialog.cpp

```diff
@@ -155,31 +147,84 @@ void OptionsDialog::on_lineEdit_server_ip_textChanged(const QString& arg1)
+    this->ui->spin_server_port->setEnabled(m_options->m_tcp_transport && mb_started);
+    this->ui->lineEdit_server_ip->setEnabled(m_options->m_tcp_transport && mb_started);
 
-	QFont font, fontn;
-	font.setBold(false);
-	fontn.setBold(true);
-	fontn.setWeight(75);
+    if (QString("TCP LAN Server") == m_options->m_tcp_type)
+    {
+        this->ui->spin_server_port->setEnabled(false);
+        this->ui->lineEdit_server_ip->setEnabled(false);
+    }
+    if (QString("TCP WAN Server") == m_options->m_tcp_type)
+    {
+        this->ui->spin_server_port->setEnabled(false);
+    }
+    if (QString("TCP Client") == m_options->m_tcp_type)
+    {
+        this->ui->spin_listen_port->setEnabled(false);
+    }
+}
+
+void OptionsDialog::on_IntraprocesscheckBox_stateChanged(int arg1)
+{
+    m_options->m_intraprocess_transport = arg1;
+    mp_sd->setOptions(*m_options);
+}
 
-	this->ui->pushButton_tcp_client->setFont(!mb_started && !m_options->m_udpTransport && !m_options->m_tcpServer ? fontn : font);
-	this->ui->pushButton_udp->setFont(!mb_started && m_options->m_udpTransport ? fontn : font);
-	this->ui->pushButton_tcp_server->setFont(!mb_started && !m_options->m_udpTransport && m_options->m_tcpServer && !m_options->m_tcpWAN ? fontn : font);
-	this->ui->pushButton_tcp_WAN_server->setFont(!mb_started && !m_options->m_udpTransport && m_options->m_tcpServer && m_options->m_tcpWAN ? fontn : font);
+void OptionsDialog::on_DataSharingcheckBox_stateChanged(int arg1)
+{
+    m_options->m_datasharing_transport = arg1;
+    mp_sd->setOptions(*m_options);
+}
 
-	
-    // lineEdit_server_ip meaning depends on TCP config.
-    this->ui->label_6->setText(m_options->m_tcpServer && m_options->m_tcpWAN ? QApplication::translate("OptionsDialog", "WAN IP:", nullptr) : QApplication::translate("OptionsDialog", "Server IP:", nullptr));
-    this->ui->lineEdit_server_ip->setEnabled(mb_started && !m_options->m_udpTransport && !(m_options->m_tcpServer && !m_options->m_tcpWAN));
-}
\ No newline at end of file
+void OptionsDialog::on_SHMcheckBox_stateChanged(int arg1)
+{
+    m_options->m_shm_transport = arg1;
+    mp_sd->setOptions(*m_options);
+    check_transports();
+}
+
+void OptionsDialog::on_UDPcheckBox_stateChanged(int arg1)
+{
+    m_options->m_udp_transport = arg1;
+    mp_sd->setOptions(*m_options);
+    check_transports();
+}
+
+void OptionsDialog::on_TCPcheckBox_stateChanged(int arg1)
+{
+    m_options->m_tcp_transport = arg1;
+    mp_sd->setOptions(*m_options);
+
+    tcp_enable_buttons();
+    check_transports();
+}
+
+void OptionsDialog::on_TCPcomboBox_currentTextChanged(const QString &arg1)
+{
+    m_options->m_tcp_type = arg1;
+    mp_sd->setOptions(*m_options);
+
+    tcp_enable_buttons();
+}
+
+void OptionsDialog::check_transports()
+{
+    if (!m_options->m_tcp_transport && !m_options->m_udp_transport && !m_options->m_shm_transport)
+    {
+        this->ui->UDPcheckBox->setChecked(mb_started);
+    }
+}
```

Comment:
When clicking twice over the `UDP` checkbox when no other option is selected, the checkbox is deselected and `create_participant` issues and error.

Reply:
Solved with new design (using default transports if none is configured).